NotationV0/src/lib/variables.ts

```diff
@@ -1,22 +1,34 @@
 import * as taskLib from 'azure-pipelines-task-lib/task';
+import { STATUS, WARNING } from './constants';
 
 // get artifact references from input or from previous docker push task
 // through "RESOURCE_URIS" variable.
 export function getArtifactReferences(): string[] {
     let artifactRefs = []
     const artifactRefsInput = taskLib.getInput('artifactRefs', false);
     if (!artifactRefsInput) {
+        console.log(taskLib.loc('TryToGetArtifactRefsFromDockerTask'));
         artifactRefs = getArtifactReferencesFromDockerTask();
-        console.log(taskLib.loc('GotArtifactRefsFromDockerTask'));
     } else {
-        artifactRefs = artifactRefsInput.split(',');
+        artifactRefs = artifactRefsInput.split(',').map((artifactRef) => artifactRef.trim());
     }
 
     if (artifactRefs.length === 0) {
         throw new Error(taskLib.loc('ArtifactRefsNotSpecified'));
     }
+
+    // validate repeated artifact references
+    let uniqueArtifactRefs = new Set<string>();
+    for (const artifactRef of artifactRefs) {
+        if (uniqueArtifactRefs.has(artifactRef)) {
+            taskLib.warning(taskLib.loc('RepeatedArtifactRef', artifactRef));
```

Comment:
Do we also need to `console.log` here? `taskLib.warning` shows the warning at the top right? `console.log` shows the warning as it generates so user could read it in order and with context? I might be wrong though, just to confirm.

Reply:
The warning log will also be shown in the logs. For example,
![image](https://github.com/AzureCR/notation-ado-task/assets/43160897/1407ac6a-4e01-4117-8519-d0858f69afbd)
The warning is shown in order as well.